From 250d9818d1eea08ff79a2457aab86f441c72b06b Mon Sep 17 00:00:00 2001
From: "Sieranski, Marek" <Marek.Sieranski@atmel.com>
Date: Fri, 23 Jan 2015 09:20:59 +0800
Subject: [PATCH] lib: fdt: fix bug: when appended a string, the total size
 fail to increase.

When a string is appended at the end of strings block within flattened device
tree, the calculated "len" is zero. The "newlen" should be used to increase
total size as it has been used with stringslen.

Reported-by: Sieranski, Marek <Marek.Sieranski@atmel.com>
Tested-by: Sieranski, Marek <Marek.Sieranski@atmel.com>
Signed-off-by: Sieranski, Marek <Marek.Sieranski@atmel.com>
---
 lib/fdt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/fdt.c b/lib/fdt.c
index d3de227..4a8b71d 100644
--- a/lib/fdt.c
+++ b/lib/fdt.c
@@ -323,7 +323,7 @@ static int of_blob_move_dt_string(void *blob, int newlen)
 	memmove(dest, point, len);
 
 	of_set_dt_strings_len(blob, stringslen);
-	of_set_dt_total_size(blob, of_get_dt_total_size(blob) + len);
+	of_set_dt_total_size(blob, of_get_dt_total_size(blob) + newlen);
 
 	return 0;
 }
-- 
1.9.1

